src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  hidden = false;
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] Instead of using a separate method `checkHealth` to filter out dead animals, consider filtering the `Animal.alive` array directly in the `bite` method after decreasing the health of the herbivore. This will avoid unnecessary method calls and make the code cleaner.